KalturaWowzaServer/src/main/java/com/kaltura/media_server/modules/AuthenticationModule.java

```diff
@@ -149,50 +165,108 @@ public void onStreamDestroy(IMediaStream stream) {
         }
     }
 
+    public void onRTPSessionCreate(RTPSession rtpSession)
+    {
+        String queryStr = rtpSession.getQueryStr();
+        String uriStr = rtpSession.getUri();
+        try {
+            WMSProperties properties = rtpSession.getProperties();
+            HashMap<String, String>  queryParameters = Utils.getRtmpUrlParameters(uriStr, queryStr);
+            if (queryParameters.containsKey(Constants.REQUEST_PROPERTY_ENTRY_ID)) {
+                logger.debug("onRTPSessionCreate - [" + queryParameters.get(Constants.REQUEST_PROPERTY_ENTRY_ID) + "]");
+            }
+
+            onClientConnect(properties, queryParameters, KalturaStreamType.RTSP);
+        } catch (Exception  e) {
+            logger.error("Entry authentication failed with url [" + uriStr + "]: " + e.getMessage());
+            rtpSession.rejectSession();
+            DiagnosticsProvider.addRejectedRTSPStream(rtpSession, e.getMessage());
+        }
+    }
+
+    public void onRTPSessionDestroy(RTPSession rtpSession)
+    {
+        try{
+            logger.debug("onRTPSessionDestroy - [" + rtpSession.getSessionId() + "]");
+            String entryId = Utils.getEntryIdFromRTPSession(rtpSession);
+            logger.info("Entry removed [" + entryId + "]");
+            KalturaEntryDataPersistence.entriesMapCleanUp();
+        }
+        catch (Exception  e){
+            logger.info("Error" + e.getMessage());
+        }
+    }
+
+
+
     class LiveStreamListener extends  MediaStreamActionNotifyBase{
+
+        private void shutdown(IMediaStream stream, String msg, StreamType streamType){
+
+            logger.error(msg);
+            IClient client = stream.getClient();
+
+            switch(streamType) {
```

Comment:
previouse code was better, if IClient is not null then it's RTMP 
if stream.getRTPStream() is not null it's RTSP